### PR TITLE
Implement event map models and response classes

### LINKED ISSUE #258: Adjust event map models and response classes for API
We don't have the API right now, but we have fake JSON. Could we adjust our model entities and response classes for the event map?

```json
{
  "status": "OK",
  "events": [
    {
      "id": "1",
      "title": {
        "ja": "[Sample] Meetup (ランチタイム)",
        "en": "[Sample] Meetup (Lunch Time)"
      },
      "i18nDesc": {
        "ja": "[通常イベント] 様々なテーマごとに集まって、一緒にランチを食べながらお話しましょう。席に限りがありますので、お弁当受け取り後お早めにお越しください。",
        "en": "[Regular Event] Let's gather around various themes and chat while having lunch together. Seats are limited, so please come early after picking up your bento."
      },
      "roomId": 48984,
      "message": {
        "ja": null,
        "en": null
      },
      "noShow": false,
      "moreDetailsUrl": null
    },
    {
      "id": "2",
      "title": {
        "ja": "[Sample] キャリア相談会",
        "en": "[Sample] Career Consultation"
      },
      "i18nDesc": {
        "ja": "[別室イベント] 様々なテーマごとに集まって、一緒にランチを食べながらお話しましょう。席に限りがありますので、お弁当受け取り後お早めにお越しください。",
        "en": "[Different Room Event] Let's gather around various themes and chat while having lunch together. Seats are limited, so please come early after picking up your bento."
      },
      "roomId": 51633,
      "message": {
        "ja": null,
        "en": null
      },
      "noShow": false,
      "moreDetailsUrl": null
    },
    {
      "id": "3",
      "title": {
        "ja": "[Sample] Meetup (ランチタイム)",
        "en": "[Sample] Meetup (Lunch Time)"
      },
      "i18nDesc": {
        "ja": "[中止イベント] 様々なテーマごとに集まって、一緒にランチを食べながらお話しましょう。席に限りがありますので、お弁当受け取り後お早めにお越しください。",
        "en": "[Cancelled Event] Let's gather around various themes and chat while having lunch together. Seats are limited, so please come early after picking up your bento."
      },
      "roomId": 48982,
      "message": {
        "ja": null,
        "en": null
      },
      "noShow": true,
      "moreDetailsUrl": null
    },
    {
      "id": "4",
      "title": {
        "ja": "[Sample] Meetup (ランチタイム)",
        "en": "[Sample] Meetup (Lunch Time)"
      },
      "i18nDesc": {
        "ja": "[時間変更イベント] 様々なテーマごとに集まって、一緒にランチを食べながらお話しましょう。席に限りがありますので、お弁当受け取り後お早めにお越しください。",
        "en": "[Time Changed Event] Let's gather around various themes and chat while having lunch together. Seats are limited, so please come early after picking up your bento."
      },
      "roomId": 51634,
      "message": {
        "ja": "時間が30分延長されました",
        "en": "The event has been extended by 30 minutes"
      },
      "noShow": false,
      "moreDetailsUrl": null
    }
  ],
  "rooms": [
    {
      "id": 51633,
      "name": {
        "ja": "Giraffe",
        "en": "Giraffe"
      },
      "sort": 2
    },
    {
      "id": 48982,
      "name": {
        "ja": "Jellyfish",
        "en": "Jellyfish"
      },
      "sort": 4
    },
    {
      "id": 48983,
      "name": {
        "ja": "Iguana",
        "en": "Iguana"
      },
      "sort": 0
    },
    {
      "id": 51634,
      "name": {
        "ja": "Flamingo",
        "en": "Flamingo"
      },
      "sort": 3
    },
    {
      "id": 48984,
      "name": {
        "ja": "Hedgehog",
        "en": "Hedgehog"
      },
      "sort": 1
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by sanao1006**
## Issue
- close #258 

## Overview (Required)
- Since the EventMap-related response classes are not yet available, I created them.

### CODE REVIEW

**Comment by takahirom** [comment]
Thanks. We already have a model `EventMapEvent`. Cloud we change it?

**Comment by sanao1006** [comment]
@takahirom 
Thanks!
I have an additional question!

Is the `isFavorite` property not included in the API response?
>  We already have a model EventMapEvent. Cloud we change it?

I tried to implement a function to convert an `EventMapResponse` to `List<EventMapEvent>`, but I can't do it because the `isFavorite` is not included in the API response.
Is there any other way to do this?

**Comment by takahirom** [comment]
I think it will be local data as the favorite sessions.
We might need to use datastore. But I think we can separate the issue. 👍

**Comment by takahirom** [comment]
Sorry. The new design won't have the favorite feature. So you don't have to worry about it.

**Comment by sanao1006** [comment]
Ok! 👍
In other words, is it correct to assume that we remove`isFavorite` from `EventMapEven`? (Sorry if this is incorrect)

**Comment by takahirom** [comment]
Yes! We don't implement the favorite feature for event map.

**Comment by sanao1006** [comment]
Sorry for repeating myself!
The `startTime` and `endTime` of `EventMapEvent` are not included in the `EventMapResponse`, how should I get them?

https://github.com/DroidKaigi/conference-app-2024/blob/7632720e833961a094681682be0f9e3811497eb5/core/data/src/commonMain/kotlin/io/github/droidkaigi/confsched/data/sessions/SessionsApiClient.kt#L6
I thought of a way to filter out id matches from the session retrieved with this function to get startsAt and endsAt and convert them to LocalDateTime, but I feel it would be inefficient...🤔

**Comment by takahirom** [comment]
We don't show the start time and the end time as well! Sorry for making you confusion.

File: core/data/src/commonMain/kotlin/io/github/droidkaigi/confsched/data/eventmap/EventMapApiClient.kt [hunk 1]
```diff
@@ -18,5 +17,17 @@ public interface EventMapApiClient {
...
+    return this.events
+        .mapNotNull { event ->
+            roomIdToNameMap[event.roomId]?.let { roomName ->
+                EventMapEvent(
+                    name = event.title.ja,
+                    roomName = roomName,
+                    description = event.i18nDesc.ja,
+                )
+            }
+        }
```
Comment (sanao1006) [inline core/data/src/commonMain/kotlin/io/github/droidkaigi/confsched/data/eventmap/EventMapApiClient.kt:31]: Currently, only Japanese is supported.
How can we support English?

Comment (takahirom) [reply core/data/src/commonMain/kotlin/io/github/droidkaigi/confsched/data/eventmap/EventMapApiClient.kt:31]: I think you can refer to TimetableItem.

Comment (takahirom) [reply core/data/src/commonMain/kotlin/io/github/droidkaigi/confsched/data/eventmap/EventMapApiClient.kt:31]: I think we can use "MultiLangText" class.👀

Comment (sanao1006) [reply core/data/src/commonMain/kotlin/io/github/droidkaigi/confsched/data/eventmap/EventMapApiClient.kt]: I see, thanks!!
I fixed it!

**Review by takahirom — APPROVED** [review]
Looks great! This is definitely a significant advancement in development!

**Comment by sanao1006** [comment]
Thanks for your review!!!!!!!!!!!!🎉🎉

**Pull request merged**